lib/engine/game/g_1824/game.rb

```diff
@@ -718,8 +898,128 @@ def float_corporation(corporation)
+              ps.spend(ps.cash, state_railway)
+            end
+            if ps.trains.any?
+              trains_transfered = transfer(:trains, ps, state_railway).map(&:name)
+              @log << "#{state_railway.name} receives the trains: #{trains_transfered}"
+            end
+
+            # Replace tokens
+            home_token = ps.tokens.first
+            city = home_token.city
+            city.remove_reservation!(ps)
+            city.delete_token!(home_token)
+            city.hex.tile.reservations.delete(ps)
+            home_token.remove!
+            if state_railway.tokens.find { |t| t.city == city }
+              @log << "#{ps.name} token in #{city.hex.name} is removed as #{state_railway.name} already have a token there"
+            else
+              @log << "#{ps.name} token in #{city.hex.name} is replaced with a #{state_railway.name} token"
+              city.place_token(state_railway, state_railway.next_token, free: true, check_tokenable: false)
+            end
+
+            @log << "#{ps.name} is now closed"
+            ps.close!
+          end
+
+          # Add player in priority order to presidency possible tie breaker
+          presidency_priority_order |= @players
+
+          floating_capital = (120 * 10) - pre_states_value
+          @bank.spend(floating_capital, state_railway)
+          capital_str = format_currency(floating_capital)
+          @log << "#{state_railway.name} receives floating capital #{capital_str} for all non-reserved shares"
+          # Check presidency according to rules
+          max_shares = @share_pool.presidency_check_shares(state_railway).values.max
+          if max_shares < 20
+            # Receivership - will not act, just fall in stock price, until someone takes the role
+            @log << "#{state_railway.name} has no president. It will decrease in value each OR until someone purchases up to 20%"
+            state_railway.owner = @share_pool
+            presidency_share = state_railway.shares.find(&:president)
+            presidency_share.buyable = false
+            # The president share will be given to the first player that has 2 10% shares
+          else
+            majority_share_holders = @share_pool.presidency_check_shares(state_railway).select { |_, p| p == max_shares }.keys
+            actual_president = majority_share_holders.max_by { |s| presidency_priority_order.find_index(s) }
+            @log << "#{actual_president.name} becomes president for #{state_railway.name}"
+            presidency_share = state_railway.presidents_share
+            if presidency_share.owner != actual_president
+              @share_pool.change_president(presidency_share, presidency_share.owner, actual_president)
+            end
+            state_railway.owner = actual_president
+          end
+          @log << "#{state_railway.name} floats and will operate next OR"
+          state_railway.floatable = true
+          state_railway.floated = true
+        end
```

Comment:
this method is way too large, can you split it into multiple smaller methods? look at other games that do mergers/nationalizations for examples